core/src/main/scala/cr/pulsar/Consumer.scala

```diff
@@ -64,7 +66,7 @@ trait Consumer[F[_], E] {
 
 object Consumer {
 
-  case class Message[A](id: MessageId, payload: A)
+  case class Message[A](id: MessageId, key: MessageKey, payload: A)
```

Comment:
Should we make `MessageKey` `Option` otherwise users will be forced to provide one even if they do not want to.

Reply:
Check the implementation of `MessageKey`